Bottom: e63a7a603f66fefd0caa239b42606c3ae0e040be
Top:    6a49cdaa7d928f7ad524af76d9ce600ac1a4b501
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 19:40:01 +0000

Refresh of allocate-vnode-with

---

diff --git a/src/polygon1.c b/src/polygon1.c
index de2ba20..961a626 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -197,10 +197,10 @@ poly_CreateNodeFull (Vector v, bool is_round, double cx, double cy, double radiu
   Coord *c;
 
   assert (v);
-  res = (VNODE *) calloc (1, sizeof (VNODE));
+//  res = (VNODE *) calloc (1, sizeof (VNODE));
+  res = g_slice_new0 (VNODE);
   if (res == NULL)
     return NULL;
-  // bzero (res, sizeof (VNODE) - sizeof(Vector));
 
 #if 0
   if (is_round)
@@ -386,6 +386,7 @@ new_descriptor (VNODE * a, char poly, char side)
                 PREV_VERTEX (a)->cvc_prev = PREV_VERTEX (a)->cvc_next = NULL;
               poly_ExclVertex (PREV_VERTEX (a));
               vect_sub (v, PREV_VERTEX (a)->point, a->point);
+#warning DOES THIS LEAK A VERTEX?
             }
           else
             {
@@ -393,6 +394,7 @@ new_descriptor (VNODE * a, char poly, char side)
                 NEXT_VERTEX (a)->cvc_prev = NEXT_VERTEX (a)->cvc_next = NULL;
               poly_ExclVertex (NEXT_VERTEX (a));
               vect_sub (v, NEXT_VERTEX (a)->point, a->point);
+#warning DOES THIS LEAK A VERTEX?
             }
         }
 
@@ -3573,7 +3575,8 @@ poly_NewContour (VNODE *node)
   res->head.radius = node->radius;
 #warning THIS WILL BE BOGUS IF WE GET A CIRCULAR CONTOUR STARTING AT THE HEAD.. NEED 2ND POINT TO DETMERMINE BOUNDS
   cntrbox_adjust (res, res->head.point);
-  free (node);
+  g_slice_free (VNODE, node);
+//  free (node);
 
   return res;
 }
@@ -3587,7 +3590,8 @@ poly_ClrContour (PLINE * c)
   while ((cur = NEXT_EDGE (&c->head)) != &c->head)
     {
       poly_ExclVertex (cur);
-      free (cur);
+      g_slice_free (VNODE, cur);
+//      free (cur);
     }
   free (c->tristrip_vertices);
   c->tristrip_vertices = NULL;
@@ -3610,7 +3614,8 @@ poly_DelContour (PLINE ** c)
 	  free (cur->cvc_next);
 	  free (cur->cvc_prev);
 	}
-      free (cur);
+      g_slice_free (VNODE, cur);
+//      free (cur);
     }
   if ((*c)->head.cvc_next != NULL)
     {
@@ -3654,7 +3659,8 @@ poly_PreContour (PLINE * C, BOOLp optimize)
           if (0)
 	    {
 	      poly_ExclVertex (c);
-	      free (c);
+	      g_slice_free (VNODE, c);
+//	      free (c);
 	      c = p;
 	    }
 	}
